scylla/src/errors.rs

```diff
@@ -354,8 +354,7 @@ impl ConnectionSetupRequestError {
 }
 
 #[derive(Error, Debug)]
-#[non_exhaustive]
-pub enum BrokenConnectionErrorKind {
+pub(crate) enum BrokenConnectionErrorKind {
     #[error("Timed out while waiting for response to keepalive request on connection to node {0}")]
     KeepaliveTimeout(IpAddr),
     #[error("Failed to execute keepalive query: {0}")]
```

Comment:
Commit: `errors: make BrokenConnectionErrorKind pub(crate) `

I don't agree with the reasoning in the commit message. I agree that using `Arc<dyn Error>` means that usually user should not care what the specific type is. Note the word "usually". Using `Arc<dyn Error>` is enough of a deterrent from using specific type - precisely because it requires downcasting and not just matching.
If however a user wants to use the specific type and match the reason they should be able to. And for that to be possible the error needs to be public (and non_exhaustive).